packages/@atjson/source-html/src/converter/social-embeds.ts

```diff
@@ -0,0 +1,124 @@
+import Document, { Annotation } from "@atjson/document";
+import OffsetSource, { IframeEmbed } from "@atjson/offset-annotations";
+import UrlSource from "@atjson/source-url";
+
+function isInstagramOrTwitterBlockquote(annotation: Annotation<any>) {
+  if (annotation.type !== "blockquote") {
+    return false;
+  }
+
+  let classes = annotation.attributes.class;
+
+  if (!classes) {
+    return false;
+  }
+
+  let classList = classes.split(" ");
+  return (
+    classList.includes("instagram-media") || classList.includes("twitter-tweet")
+  );
+}
+
+function isFacebookDiv(annotation: Annotation<any>) {
+  if (annotation.type !== "div") {
+    return false;
+  }
+
+  return (
+    annotation.attributes.dataset &&
+    annotation.attributes.dataset.href &&
+    new URL(annotation.attributes.dataset.href).host === "www.facebook.com"
+  );
+}
+
+export default function(doc: Document) {
+  doc.where({ type: "-html-iframe" }).update(iframe => {
+    let { height, width, src: url } = iframe.attributes;
+    let urlDoc = UrlSource.fromRaw(url).convertTo(OffsetSource);
+    let embed;
+    if (urlDoc.annotations.length) {
+      embed = urlDoc.annotations[0];
+      embed.start = iframe.start;
+      embed.end = iframe.end;
+    } else {
+      embed = new IframeEmbed({
+        start: iframe.start,
+        end: iframe.end,
+        attributes: {
+          url
+        }
+      });
+    }
+    embed.attributes = { ...embed.attributes, height, width };
```

Comment:
This was a bit hard for me to track, but I think I got it. Is there any cases where we _may_ not get a height and width on these? (If so, what is the expected behavior?)

Reply:
In those cases height and width will be undefined which seems reasonable since height and width are optional attributes